kytos/core/user.py

```diff
@@ -0,0 +1,93 @@
+"""User authentification """
+import hashlib
+from datetime import datetime
+from typing import Literal, Optional
+
+# pylint: disable=no-name-in-module
+from pydantic import BaseModel, EmailStr, Field, constr, validator
+
+
+class DocumentBaseModel(BaseModel):
+    """DocumentBaseModel"""
+
+    id: str = Field(None, alias="_id")
+    inserted_at: Optional[datetime]
+    updated_at: Optional[datetime]
+    deleted_at: Optional[datetime]
+
+    def dict(self, **kwargs) -> dict:
+        """Model to dict."""
+        values = super().dict(**kwargs)
+        if "id" in values and values["id"]:
+            values["_id"] = values["id"]
+        if "exclude" in kwargs and "_id" in kwargs["exclude"]:
+            values.pop("_id")
+        return values
+
+
+class UserDoc(DocumentBaseModel):
+    """UserDocumentModel."""
+
+    username: str
```

Comment:
Right now it is possible to create a user with an empty username. Would be a good idea to add a constraint like `min_length=1`?.

Reply:
Good catch, @Alopalao, let's go for it, and then in the future we can keep incrementing the validation as needed.